src/profile.jl

```diff
@@ -0,0 +1,321 @@
+
+Replace `sink[i]` by `max(source[i], bd[i] + incr)`.  When `bd[i] == -Inf` this simply copies `source[i]`.
+"""
+function _copy_away_from_lowerbd!(sink, source, bd; incr=0.01)
+    for i in eachindex(sink, source, bd)
+        @inbounds sink[i] = max(source[i], bd[i] + incr)
+    end
+    return sink
+end
+
+function σvals(m::LinearMixedModel{T}) where {T}
+    @compat (; σ, reterms) = m
+    isone(length(reterms)) && return σvals(only(reterms), σ)
+    return (collect(Iterators.flatten(σvals.(reterms, σ)))...,)
+end
+
+function FeProfile(m::LinearMixedModel, j::Integer)
+    Xy = m.Xymat.xy
+    xcols = collect(axes(Xy, 2))
+    ycol = pop!(xcols)
+    notj = deleteat!(xcols, j) # indirectly check that j ∈ xcols
+    y₀ = Xy[:, ycol]
+    xⱼ = Xy[:, j]
+    feterm = FeTerm(Xy[:, notj], m.feterm.cnames[notj])
+    reterms = [copy(ret) for ret in m.reterms]
+    mnew = fit!(LinearMixedModel(y₀ - xⱼ * m.β[j], feterm, reterms, m.formula); progress=false)
+       # not sure this next call makes sense - should the second argument be m.optsum.final?
+    _copy_away_from_lowerbd!(mnew.optsum.initial, mnew.optsum.final, mnew.lowerbd; incr=0.05)
+    return FeProfile(mnew, y₀, xⱼ, j)
+end
+
+function betaprofile!(pr::FeProfile{T}, val, βⱼ::T, j::Integer, obj::T) where {T}
+    prm = pr.m
+    refit!(prm, mul!(copyto!(pr.m.y, pr.y₀), pr.xⱼ, βⱼ, -1, 1); progress=false)
+    ζ = sqrt(prm.objective - obj)
+    push!(val.ζ, ζ)
+    push!(val.θ, (prm.θ...,))
+    push!(val.σ, (prm.σ, σvals(prm)...))
+    β = prm.β
+    splice!(β, j:(j - 1), βⱼ)
+    push!(val.β, (β...,)) # FIXME: do this in a loop rather than splice! followed by splatting
+    return ζ
+end
+
+function profileβj(m::LinearMixedModel{T}, j::Integer; threshold=4) where {T}
+    @compat (; β, θ, σ, stderror, objective) = m
+    val = (; ζ = -T[0], β = [(β...,)], σ = [(σ, σvals(m)...)], θ = [(θ...,)])
+    prj = FeProfile(m, j)
+    st = stderror[j] * 0.5
+    bb = β[j] - st
+    while true
+        betaprofile!(prj, val, bb, j, objective) < threshold || break
+        bb -= st
+    end
+    rmul!(val.ζ, -1)
+    bb = β[j] + st
+    while true
+        betaprofile!(prj, val, bb, j, objective) < threshold || break
+        bb += st
+    end
+    m = length(val.ζ)
+    return sort!(Table(merge((; p = fill(:β, m), j = fill(Int8(j), m)), val)), by = r -> r.ζ)
+end
+
+struct MixedModelProfile
+    m::LinearMixedModel   # Model that has been profiled
+    tbl::Table            # Table containing ζ, σ, β, and θ from each conditional fit
+    fwd::Dict{Symbol}     # Interpolation splines for ζ as a function of a parameter
+    rev::Dict{Symbol}     # Interpolation splines for a parameter as a function of ζ
+end
+
+"""
+    profile(m::LinearMixedModel)
+
+Return a `MixedModelProfile` for the objective of `m` with respect to the fixed-effects coefficients.
+"""
+function profile(m::LinearMixedModel{T}; threshold = 4) where {T}
+    final = copy(refit!(m).optsum.final)
```

Comment:
Is the `refit!` necessary?

Reply:
Just being extra-cautious while developing the code.  Past-me has a habit of leaving the model object in a corrupt state.